e2e/framework/provisioning/default.go

```diff
@@ -55,8 +45,8 @@ func (p *singleClusterProvisioner) ProvisionCluster(opts ProvisionerOptions) (*C
 
 	if opts.ExpectConsul {
 		consulClient, err := capi.NewClient(capi.DefaultConfig())
-		if err != nil && opts.ExpectConsul {
-			return nil, err
+		if err != nil {
+			return nil, fmt.Errorf("expected Consul: %v", err)
```

Comment:
i really like `errors.Wrapf(err, "expected Consul")` (from `github.com/pkg/errors`)

Reply:
I don't have a strong opinion on it, but it looks like we've avoided that in all but one spot in the Nomad codebase, so I'd like to keep the existing idiom if we can? IIRC there's some cool new error wrapping features in a newer golang, so maybe there's a codebase-wide improvement to be had once we get bumped to the right minimum version.